db/memtable_list.cc

```diff
@@ -752,14 +746,21 @@ Status InstallMemtableAtomicFlushResults(
     edit_lists.emplace_back(edits);
   }
 
-  // TODO(cc): after https://github.com/facebook/rocksdb/pull/7570, handle 2pc
-  // here.
+  WalNumber min_wal_number_to_keep = 0;
+  if (vset->db_options()->allow_2pc) {
+    min_wal_number_to_keep = PrecomputeMinLogNumberToKeep2PC(
+        vset, cfds, edit_lists, mems_list, prep_tracker);
+    edit_lists.back().back()->SetMinLogNumberToKeep(min_wal_number_to_keep);
+  }
+
   std::unique_ptr<VersionEdit> wal_deletion;
-  if (vset->db_options()->track_and_verify_wals_in_manifest) {
-    uint64_t min_wal_number_to_keep =
-        PrecomputeMinLogNumberToKeepNon2PC(vset, cfds, edit_lists);
-    const auto& wals = vset->GetWalSet().GetWals();
-    if (!wals.empty() && min_wal_number_to_keep > wals.begin()->first) {
+  if (vset->db_options()->track_and_verify_wals_in_manifest &&
+      !vset->GetWalSet().GetWals().empty()) {
+    if (!vset->db_options()->allow_2pc) {
+      min_wal_number_to_keep =
+          PrecomputeMinLogNumberToKeepNon2PC(vset, cfds, edit_lists);
+    }
+    if (min_wal_number_to_keep > vset->GetWalSet().GetWals().begin()->first) {
```

Comment:
Can we do something like:
```
WalNumber min_wal_number_to_keep = allow_2pc ? PrecomputeMinLogNumberToKeep2PC() : PrecomputeMinLogNumberToKeepNon2PC()?
edit.SetMinLogNumberToKeep(min_wal_ number_to_keep);
```
I guess I am not sure I miss anything when I see `min_wal_number_to_keep` is evaluated in allow_2pc mode but only conditionally in non-2pc mode. In either case, it would be OK for the db to track min_wal_number_to_keep, right?

Reply:
I agree that it's a bit confusing. But according to https://github.com/facebook/rocksdb/blob/master/db/version_set.cc#L4254, the min_log_number_to_keep in VersionEdit now should only be set in 2pc mode.